macos/FluentUI/Button/Button.swift

```diff
@@ -388,125 +388,77 @@ class ButtonCell: NSButtonCell {
 			return NSZeroRect
 		}
 
-		let isRTL = NSApp.userInterfaceLayoutDirection == .rightToLeft
-		let imageRect = super.imageRect(forBounds: rect)
+		let layoutDirectionSign : CGFloat = NSApp.userInterfaceLayoutDirection == .rightToLeft ? -1 : 1
 		let titleSize = title.size(withAttributes: [.font: font as Any])
 		let imageSize = image.size
 
-		var x = imageRect.origin.x
-		var y = imageRect.origin.y
-		var width = imageRect.size.width
-		var height = imageRect.size.height
+		var x = (rect.width - imageSize.width) / 2
+		var y = (rect.height - imageSize.height) / 2
+
+		let xOffsetFromTitle = (titleSize.width + titleToImageSpacing) / 2
+		let yOffsetFromTitle = (titleSize.height + titleToImageSpacing - titleToImageVerticalSpacingAdjustment) / 2
 
 		switch imagePosition {
 		case .noImage:
-			x = 0
-			y = 0
-			width = 0
-			height = 0
+			return NSZeroRect
```

Comment:
nit: can we avoid the early return in this one case? That would require introducing two vars `width` and `height` that default to `imageSize.width` and `imageSize.height` which is arguably less clean, so I don't feel too strongly. 

The preference for no early returns comes from our [obj-s guide](https://microsoft.github.io/objc-guide/MethodsAndImplementations/EarlyReturns.html) and our [swift guide](https://microsoft.github.io/swift-guide/EarlyReturn.html).

Reply:
What's bad about early return?  Would it be better if it was even earlier?

Reply:
Literally no other case uses zero for any of the four values: this case is a total outlier in terms of the values.

Reply:
The intermediate variables improve readability and reduce the chance of mistakes since I am not writing multiple long lines of math for each case.  I am not too worried about the performance implications of a couple of CGFloat calculations done upfront in the service of a much simpler switch block.

Reply:
> choosing a reasonable default for all values that need to be set

Out of 9 cases, 8 have the same width and height values.  4 have the same x values, and 4 of the other 5 are those same values plus or minus a constant.  6 have the same y values, and 2 of the other 3 are those same values plus or minus a constant.  It seems to me the choice of initial values is pretty optimal...

Reply:
Then break that case out into a separate logical block or break handling from the other cases into a separate func? As mentioned in the links Saad shared, early returns are often used as a crutch where different factoring would improve clarity - imagine I'm investigating a bug in the no image case and I'm debugging through and trying to understand why we're doing calculations for things that aren't used and trying to understand whether that's by design or a code bug.

In ObjC/Swift I've not seen a case (other than guard lets for input validation) where adding early returns actually improves code clarity relative to more explicit factoring. I've seen many cases where it causes bugs due to control flow being more difficult to parse with stuff like added cleanup surprisingly not applying to all code paths.

Reply:
Moved the early return to the `guard` clause at top.  Not sure if the preferred style is to use one or many of these, but they all return the same thing so I used one.

Reply:
Introduced a couple of new 'sign' vars to track +/-/0 until the end of the switch block where I can determine whether to compute the actual offset values.  I still left these as mutating x/y to keep line lengths reasonable and math semi-intelligible.